extensions/cache/deployment/src/main/java/io/quarkus/cache/deployment/CacheProcessor.java

```diff
@@ -74,14 +75,14 @@ ValidationErrorBuildItem validateBeanDeployment(ValidationPhaseBuildItem validat
     }
 
     @BuildStep
-    @Record(STATIC_INIT)
-    void recordCachesBuild(CombinedIndexBuildItem combinedIndex, BeanContainerBuildItem beanContainer, CacheConfig config,
-            CaffeineCacheBuildRecorder caffeineRecorder) {
+    @Record(RUNTIME_INIT)
```

Comment:
Why did you move to runtime init?

Reply:
When this PR was created, the generated bytecode execution time was `STATIC_INIT`, but it's no longer true on `master` today. I had to move it to `RUNTIME_INIT` on `master` to allow context propagation into cache values computations (see #8227). In this PR, this move was caused by the introduction of the executor provided by `ExecutorBuildItem` (see `CacheProcessor`).

Reply:
Damn that sucks. Anything we can move the needle on @FroMage @mkouba ?

Reply:
We found a solution for this, `RUNTIME_INIT` was replaced with `STATIC_INIT` a while ago on `master`.